### PR TITLE
c-api 导出 适配不同的编译器

### PR DESCRIPTION
**Pull request by zhou881117**
#if defined(__BORLANDC__)
    #if defined(__clang__)
        #ifdef __cplusplus
           #define SherpaNcnn_EXPORT extern "C"
        #else
           #define SherpaNcnn_EXPORT
        #endif
    #else
        #error Cannot define PACKED macros for this compiler
    #endif
#elif defined(_MSC_VER)
    #ifdef __cplusplus
           #define SherpaNcnn_EXPORT   extern "C" _declspec(dllexport)
    #else
       #define SherpaNcnn_EXPORT
   #endif

#elif defined(__GNUC__)
    #ifdef __cplusplus
          #define SherpaNcnn_EXPORT extern "C"
    #else
          #define SherpaNcnn_EXPORT
    #endif
#else
    #error PACKED macros are not defined for this compiler
#endif

### CODE REVIEW

File: sherpa-ncnn/c-api/c-api.h [hunk 1]
```diff
@@ -27,6 +27,35 @@
...
+
+#if defined(__BORLANDC__)
+    #if defined(__clang__)
+        #ifdef __cplusplus
+           #define SherpaNcnn_EXPORT extern "C"
+        #else
+           #define SherpaNcnn_EXPORT
+        #endif
+    #else
+        #error Cannot define PACKED macros for this compiler
```
Comment (csukuangfj) [inline sherpa-ncnn/c-api/c-api.h:39]: What does PACKED mean in this context?

Comment (zhou881117) [reply sherpa-ncnn/c-api/c-api.h:39]: #if defined(_MSC_VER)
    #ifdef __cplusplus
        #define SherpaNcnn_EXPORT   extern "C" _declspec(dllexport)
    #else
       #define SherpaNcnn_EXPORT
   #endif
#else
   #ifdef __cplusplus
       #define SherpaNcnn_EXPORT extern "C"
   #else
        #define SherpaNcnn_EXPORT
   #endif
#endif

这个项目，这样应该就ok了。
区分msvc就好

Comment (csukuangfj) [reply sherpa-ncnn/c-api/c-api.h:39]: 谢谢。可以提个Pull request 吗？

Comment (zhou881117) [reply sherpa-ncnn/c-api/c-api.h:39]: 已经重新提交了

**Pull request closed without merging**